crates/binjs_io/src/entropy/predict.rs

```diff
@@ -336,3 +346,172 @@ impl<NodeValue> PathPredict<NodeValue, SymbolInfo> where NodeValue: Eq + Hash +
     }
 }
 
+
+
+
+
+#[derive(Debug, PartialEq, Eq, Hash, Clone, Serialize, Deserialize)]
+enum WindowPrediction {
+    BackReference(BackReference),
+    DictionaryIndex(DictionaryIndex),
+}
+
+
+#[derive(Clone, Debug, Serialize, Deserialize)]
+pub struct WindowPredict<NodeValue, Statistics> where NodeValue: Clone + Eq + Hash {
+    /// The window width
+    width: usize,
+
+    /// All the known values.
+    value_by_dictionary_index: Vec<NodeValue>,
+    dictionary_index_by_value: HashMap<NodeValue, DictionaryIndex>,
+
+    latest_values: Vec<NodeValue>,
```

Comment:
Please add comment what the entries are:
  * the order inside this vector
    * which side is the first
  * where the current index points inside the vector
  * what the current index is
  * where "back reference" points